### PR TITLE
Backports 0.18 pr19

### PR DESCRIPTION
**Pull request by Munkybooty**
There were many test failures that occurred due to #15419, i managed to significantly reduce the amount. However some functional tests still fail. I'm not sure what remains to be fixed.

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see below + https://github.com/UdjinM6/dash/commits/pr4642

File: src/Makefile.test.include [hunk 1]
```diff
@@ -512,6 +520,13 @@ if BUILD_BITCOIN_TX
 endif
 	@echo "Running test/util/rpcauth-test.py..."
 	$(PYTHON) $(top_builddir)/test/util/rpcauth-test.py
+if TARGET_WINDOWS
+else
+if ENABLE_BENCH
+	@echo "Running bench/bench_dash"
+	$(BENCH_BINARY)
```
[the remark this answers is not in the document]
Comment (Munkybooty) [reply src/Makefile.test.include:527]: `$(BENCH_BINARY) -evals=1 -scaling=0 > /dev/null` had continually thrown errors at me I'm not sure what to do in that regard

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/test_framework/test_framework.py [hunk 2]
```diff
@@ -366,17 +366,14 @@ def setup_nodes(self):
...
-
         for n in self.nodes:
             try:
                 n.getwalletinfo()
             except JSONRPCException as e:
                 assert str(e).startswith('Method not found')
                 continue
 
-            n.importprivkey(n.get_deterministic_priv_key().key)
+            n.importprivkey(privkey=n.get_deterministic_priv_key().key, label='coinbase')
```
[the remark this answers is not in the document]
Comment (UdjinM6) [reply test/functional/test_framework/test_framework.py:364]: Still not fixed. I'd suggest dropping 14631 for now and backporting it together with 14468 sometimes later.

File: src/rpc/rawtransaction.cpp [hunk 3]
```diff
@@ -114,21 +114,21 @@ void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry)
...
-            "\nHint: use getmempoolentry to fetch a specific transaction from the mempool.\n"
+                "\nHint: use getmempoolentry to fetch a specific transaction from the mempool.\n"
                 "Or use gettransaction for wallet transactions.\n"
-            "\nIf verbose is 'true', returns an Object with information about 'txid'.\n"
-            "If verbose is 'false' or omitted, returns a string that is serialized, hex-encoded data for 'txid'.\n",
+
+                "\nHint: Use gettransaction for wallet transactions.\n"
+
+                "\nIf verbose is 'true', returns an Object with information about 'txid'.\n"
+                "If verbose is 'false' or omitted, returns a string that is serialized, hex-encoded data for 'txid'.\n",
```
Comment (UdjinM6) [inline src/rpc/rawtransaction.cpp:131]: 15477: this looks different in the original PR

File: src/Makefile.bench.include [hunk 4]
```diff
@@ -78,7 +78,7 @@ bench_bench_dash_SOURCES += bench/coin_selection.cpp
 bench_bench_dash_SOURCES += bench/wallet_balance.cpp
 endif
 
-bench_bench_dash_LDADD += $(BACKTRACE_LIB) $(BOOST_LIBS) $(BDB_LIBS) $(EVENT_PTHREADS_LIBS) $(EVENT_LIBS) $(MINIUPNPC_LIBS) $(NATPMP_LIBS) $(EVENT_PTHREADS_LIBS) $(EVENT_LIBS) $(BLS_LIBS) $(GMP_LIBS)
+bench_bench_dash_LDADD += $(BACKTRACE_LIB) $(BOOST_LIBS) $(BDB_LIBS) $(MINIUPNPC_LIBS) $(NATPMP_LIBS) $(BLS_LIBS) $(GMP_LIBS)
```
Comment (UdjinM6) [inline src/Makefile.bench.include:81]: 14212: should only drop the second pair, the first one was added via 15288 and should be kept i.e.
```suggestion
bench_bench_dash_LDADD += $(BACKTRACE_LIB) $(BOOST_LIBS) $(BDB_LIBS) $(EVENT_PTHREADS_LIBS) $(EVENT_LIBS) $(MINIUPNPC_LIBS) $(NATPMP_LIBS) $(BLS_LIBS) $(GMP_LIBS)
```

File: src/qt/bitcoingui.cpp [hunk 5]
```diff
@@ -1756,7 +1756,7 @@ bool BitcoinGUI::handlePaymentRequest(const SendCoinsRecipient& recipient)
     return false;
 }
 
-void BitcoinGUI::setHDStatus(int hdEnabled)
+void BitcoinGUI::setHDStatus(bool privkeyDisabled, int hdEnabled)
```
Comment (UdjinM6) [inline src/qt/bitcoingui.cpp:1759]: 13966: https://github.com/dashpay/dash/pull/4642#discussion_r811417386

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by UdjinM6** [comment]
pls see 2d91636a83 and 0423c5dfdd

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**